sycl/doc/extensions/SPIRV/SPV_INTEL_vector_compute.asciidoc

```diff
@@ -0,0 +1,199 @@
+Version 1.5, Revision 2, Unified
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds new capability, execution mode and decorations
+to support INTEL vector compute specific features in SPIR-V.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_vector_compute"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 2, Unified
+---------------------------------------------------------------------------
+
+Modify Section 2.2.2, Types, add *OpTypeBufferSurfaceINTEL* to the list of _Opaque Types_
+
+Modify Section 3.6, Execution Mode, add the following rows to the Execution Mode table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Execution Mode  | Enabling Capabilities | Extra Operands
+| 5618 | *SharedLocalMemorySizeINTEL* +
+Specifies the size of shared memory required by kernel, in bytes.
+
+'Size' is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Size_
+|====
+--
+Modify Section 3.7, Storage Class: +
+Add *VectorComputeINTEL* to the list of enabling capabilities for the *Private* storage class.
+
+Modify Section 3.20, Decoration, add the following rows to the Decoration table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Decoration  | Enabling Capabilities | Extra Operands
+| 5626 | *VectorComputeFunctionINTEL* +
+Apply to a function definition to specify that the function is a vector compute function.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* |
+| 5624 | *VectorComputeVariableINTEL* +
+Apply to a variable to specify that the variable is a vector compute variable.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* |
+| 5627 | *StackCallINTEL* +
+Apply to a function definition to specify that this function is a stack call function.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* |
+| 5599 | *SIMTCallINTEL* +
+Apply to a function definition to specify that this function must be vectorized to width specified in 'N'.
+
+'N' is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _N_
+| 5625 | *FuncParamIOKindINTEL* +
+Apply to a function parameter to specify properties of the register file corresponding to the parameter.
+
+'Kind' is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Kind_
+| 5628 | *GlobalVariableOffsetINTEL* +
+Apply to a variable to specify byte offset in general register file.
+
+'Offset' is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Offset_
+| 6085 | *SingleElementVectorINTEL* +
```

Comment:
We already have `VectorAnyINTEL` that allows us to have single-element vectors, why can't we update our FE to actually emit vectors of size 1 instead of scalars if we want to use such vectors?

Reply:
OpTypeVector has a limitation in the SPIR-V specification - vector size must be at least 2.